tests/components/canary/test_sensor.py

```diff
@@ -1,203 +1,175 @@
-        assert sensor.state == 1.0
-        assert sensor.icon == "mdi:weather-windy"
-
-        air_quality = sensor.device_state_attributes[ATTR_AIR_QUALITY]
-        assert air_quality == STATE_AIR_QUALITY_NORMAL
-
-    def test_air_quality_sensor_with_none_sensor_value(self):
-        """Test air quality sensor."""
-        device = mock_device(10, "Family Room", "Canary Pro")
-        location = mock_location("Home")
-
-        data = Mock()
-        data.get_reading.return_value = None
-
-        sensor = CanarySensor(data, SENSOR_TYPES[2], location, device)
-        sensor.update()
-
-        assert sensor.state is None
-        assert sensor.device_state_attributes is None
-
-    def test_battery_sensor(self):
-        """Test battery sensor."""
-        device = mock_device(10, "Family Room", "Canary Flex")
-        location = mock_location("Home")
-
-        data = Mock()
-        data.get_reading.return_value = 70.4567
-
-        sensor = CanarySensor(data, SENSOR_TYPES[4], location, device)
-        sensor.update()
-
-        assert sensor.name == "Home Family Room Battery"
-        assert sensor.unit_of_measurement == PERCENTAGE
-        assert sensor.state == 70.46
-        assert sensor.icon == "mdi:battery-70"
-
-    def test_wifi_sensor(self):
-        """Test battery sensor."""
-        device = mock_device(10, "Family Room", "Canary Flex")
-        location = mock_location("Home")
-
-        data = Mock()
-        data.get_reading.return_value = -57
-
-        sensor = CanarySensor(data, SENSOR_TYPES[3], location, device)
-        sensor.update()
-
-        assert sensor.name == "Home Family Room Wifi"
-        assert sensor.unit_of_measurement == "dBm"
-        assert sensor.state == -57
-        assert sensor.icon == "mdi:wifi"
+from homeassistant.const import ATTR_UNIT_OF_MEASUREMENT, PERCENTAGE, TEMP_CELSIUS
+from homeassistant.setup import async_setup_component
+
+from . import mock_device, mock_location, mock_reading
+
+from tests.async_mock import patch
+from tests.common import mock_registry
+
+
+async def test_sensors_pro(hass, canary) -> None:
+    """Test the creation and values of the sensors for Canary Pro."""
+    await async_setup_component(hass, "persistent_notification", {})
+
+    registry = mock_registry(hass)
+    online_device_at_home = mock_device(20, "Dining Room", True, "Canary Pro")
+
+    instance = canary.return_value
+    instance.get_locations.return_value = [
+        mock_location(100, "Home", True, devices=[online_device_at_home]),
+    ]
+
+    instance.get_latest_readings.return_value = [
+        mock_reading("temperature", "21.12"),
+        mock_reading("humidity", "50.46"),
+        mock_reading("air_quality", "0.59"),
+    ]
+
+    config = {DOMAIN: {"username": "test-username", "password": "test-password"}}
+    with patch("homeassistant.components.canary.CANARY_COMPONENTS", ["sensor"]):
+        assert await async_setup_component(hass, DOMAIN, config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)